azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/AzureAppService.java

```diff
@@ -37,56 +50,108 @@ public IWebApp webapp(String id) {
+                .collect(Collectors.toList());
     }
 
     public IWebAppDeploymentSlot deploymentSlot(String id) {
         return deploymentSlot(WebAppDeploymentSlotEntity.builder().id(id).build());
     }
 
     public IWebAppDeploymentSlot deploymentSlot(String resourceGroup, String appName, String slotName) {
-        return deploymentSlot(WebAppDeploymentSlotEntity.builder().resourceGroup(resourceGroup).webappName(appName).name(slotName).build());
+        return deploymentSlot(null, resourceGroup, appName, slotName);
+    }
+
+    public IWebAppDeploymentSlot deploymentSlot(String subscriptionId, String resourceGroup, String appName, String slotName) {
+        return deploymentSlot(WebAppDeploymentSlotEntity.builder().subscriptionId(subscriptionId).resourceGroup(resourceGroup).webappName(appName).name(slotName).build());
     }
 
     public IWebAppDeploymentSlot deploymentSlot(WebAppDeploymentSlotEntity deploymentSlot) {
-        return new WebAppDeploymentSlot(deploymentSlot, this);
+        final String subscriptionId = getSubscriptionId(deploymentSlot.getId(), deploymentSlot.getSubscriptionId());
+        return new WebAppDeploymentSlot(deploymentSlot, getAzureResourceManager(subscriptionId));
+    }
+
+    // todo: share codes with other library which leverage track2 mgmt sdk
+    @Cacheable(cacheName = "AzureResourceManager", key = "$subscriptionId")
+    private AzureResourceManager getAzureResourceManager(String subscriptionId) {
+        final Account account = Azure.az(AzureAccount.class).account();
+        final AzureConfiguration config = Azure.az().config();
+        final String userAgent = config.getUserAgent();
+        final HttpLogDetailLevel logDetailLevel = config.getLogLevel() == null ?
+                HttpLogDetailLevel.NONE : HttpLogDetailLevel.valueOf(config.getLogLevel().name());
+        final AzureProfile azureProfile = new AzureProfile(account.getEnvironment());
+        return AzureResourceManager.configure()
+                .withLogLevel(logDetailLevel)
+                .withPolicy(getUserAgentPolicy(userAgent)) // set user agent with policy
+                .authenticate(account.getTokenCredentialForSubscription(subscriptionId), azureProfile)
+                .withSubscription(subscriptionId);
+    }
+
+    private HttpPipelinePolicy getUserAgentPolicy(String userAgent) {
+        return (httpPipelineCallContext, httpPipelineNextPolicy) -> {
+            final String previousUserAgent = httpPipelineCallContext.getHttpRequest().getHeaders().getValue("User-Agent");
+            httpPipelineCallContext.getHttpRequest().setHeader("User-Agent", String.format("%s %s", userAgent, previousUserAgent));
+            return httpPipelineNextPolicy.process();
+        };
     }
 
-    public AzureResourceManager getAzureResourceManager() {
-        return azureResourceManager;
+    private String getSubscriptionId(@Nullable String resourceId, @Nullable String subscriptionId) {
```

Comment:
getDefaultSubscriptionId

Reply:
As we discussed before, entity should include subscription data, no need to refer the default value, so keep its origin name ,just try get subscription from resource id then subscription, and throw exception in other cases